pytorch_lightning/trainer/optimizers.py

```diff
@@ -187,10 +202,13 @@ def _configure_schedulers(
                     lr_schedulers.append(
                         {**default_config, "scheduler": scheduler, "reduce_on_plateau": True, "monitor": monitor}
                     )
-                elif isinstance(scheduler, optim.lr_scheduler._LRScheduler):
-                    lr_schedulers.append({**default_config, "scheduler": scheduler})
                 else:
-                    raise ValueError(f'The provided lr scheduler "{scheduler}" is invalid')
+                    lr_schedulers.append({**default_config, "scheduler": scheduler})
+
+            current_scheduler = lr_schedulers[-1]["scheduler"]
```

Comment:
Why `[-1]` and not in the loop?

Reply:
schedulers config is being created differently under different conditions up there. So check whether it's a valid scheduler or not inside every condition might make it too verbose. here is will just check the recently added one in the list.